File: Python/quick_sort.py
from typing import List


def quick_sort(data: List[int]) -> List[int]:
    """

    :type data: List[int]
    """
    if len(data) == 0:
        return []
    else:
        head, *tail = data
        smaller = quick_sort(list(filter(lambda x: x <= head, tail)))
        greater = quick_sort(list(filter(lambda x: x >= head, tail)))
        return list(smaller + [head] + greater)


print(quick_sort([5, 4, 2, 1, 77, 8, 555, 44444]))

File: kotlin/quicksort.kt
    fun quickSort(array:List<Int>):List<Int>{
        if(array.size == 0){
            return listOf()
        }
        else{
            val head = array.first()
            val tail = array.subList(1, array.size - 1)
            val smaller = quickSort(tail.filter { x -> head > x })
            val greater = quickSort(tail.filter { x -> head <= x })

            return smaller.plus(head).plus(greater)
        }
    }
    fun main(args : Array<String>) {
        val array = arrayListOf(1,5,7,89,77,5,1,66,444)
        val list = quickSort(array)
        print(list)
    }
